pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -172,6 +172,10 @@ func isRequiredConfigPresent(config []byte) error {
 
 	requiredPaths := [][]string{
 		{"extendedArguments", "cluster-name"},
+
+		// featuregates are needed to ensure proper security is enabled.
+		{"extendedArguments", "feature-gates"},
+		{"featureGates"},
```

Comment:
Two different variants exists due to https://github.com/openshift/cluster-kube-controller-manager-operator/pull/695. But they are done at the same time?

Reply:
they each belong to a different component, see the comments in `pkg/operator/configobservation/configobservercontroller/observe_config_controller.go`